src/main/obsRecorder.ts

```diff
@@ -135,19 +135,54 @@ const setupScene = (monitorIndex: number) => {
   console.info("monitorIndexFromZero:", monitorIndexFromZero);
   const { physicalWidth, physicalHeight } = displayInfo(monitorIndexFromZero);
 
+  const obsSettings = osn.NodeObs.OBS_settings_getSettings('Video').data;
+
+  // Checks if val is +-1 compare
+  const isClose = (val: number, compare: number) => {
+    return Math.abs(compare - val) <= 1;
+  };
+
+  // Checks if resolution is close enough to monitor, and if so, sets the settingString.
+  const checkRes = (resolution: string) => {
+    const [resW, resH] = resolution.split('x');
+
+    if (
+      isClose(parseInt(resW, 10), physicalWidth) &&
+      isClose(parseInt(resH, 10), physicalHeight)
+    ) {
+      return true;
+    }
+
+    return false;
+  };
+
+  // TODO: Output should eventually be moved into a setting field to be scaled down. For now it matches the monitor resolution.
+  obsSettings.forEach((subCategory: any) => {
+    subCategory.parameters.forEach((param: any) => {
+      if (param.name === 'Base' || param.name === 'Output') {
```

Comment:
It's usually better to either return early or use `.filter()` to remove the unwanted elements than to indent everything one more level. Fx.:

```typescript
obsSettings.forEach((subCategory: any) => {
  subCategory.parameters.forEach((param: any) => {
    if (!(param.name === 'Base' || param.name === 'Output')) {
      return;
    }

    console.log(param.name, param.currentValue, param.values);
    param.values.forEach((vpair: { [key: string]: string }) => {
      const res = Object.keys(vpair)[0];
      if (checkRes(res)) {
        param.currentValue = res;
        console.info('Setting Video', param.name, 'to', res);
      }
    });
  });
});
```

Reply:
By the way, could this not easily be implemented using the existing methods to avoid duplicating the settings code?
```typescript
getAvailableValues (category: any, subcategory: any, parameter: any);
setSetting (category: any, parameter: any, value: any);
```

Reply:
I wrote the code to do everything in the same loop, rather than looping 4x to find, set, find, set when the functions are used

Reply:
I'd argue that the reduction of duplicate code outweighs the performance penalty, which is neglible, but ultimately it's up to @aza547. :)

As long as it's not a method that is called hundreds of times per second, it's not a problem.

Reply:
I would prefer this to call to setSetting - even if it sacrifices a bit of efficiency, I don't expect this to be a bottleneck.

Reply:
@aza547 updated

> I would prefer this to call to setSetting - even if it sacrifices a bit of efficiency, I don't expect this to be a bottleneck.